src/Network/Minio/Data.hs

```diff
@@ -208,15 +210,18 @@ mkHeaderFromMetadata :: [(Text, Text)] -> [HT.Header]
 mkHeaderFromMetadata = map (\(x, y) -> (mk $ encodeUtf8 $ addXAmzMetaPrefix $ T.toLower x, encodeUtf8 y))
 
 pooToHeaders :: PutObjectOptions -> [HT.Header]
-pooToHeaders poo = userMetadata ++ zip names values
+pooToHeaders poo = userMetadata
+                   ++ map (\(k, v) ->
+                             (k, fromJust v))
+                   (filter (\(_, v) -> isJust v) (zipWith (,) names values))
   where
     userMetadata = mkHeaderFromMetadata $ pooUserMetadata poo
 
     names = ["content-type",
              "content-encoding",
              "content-disposition",
              "cache-control"]
-    values = mapMaybe (fmap encodeUtf8 . (poo &))
+    values = map (fmap encodeUtf8 . (poo &))
```

Comment:
Ah! Good catch! I suggest we add a unit test to this (perhaps even a quick check test) and avoid partial functions like from just.

Reply:
This PR has a unit test that will catch behaviour similar to what this PR fixes - https://github.com/minio/minio-hs/pull/81/files#diff-484a3971f1f07ab9aae0ed00ca751253R342
If you observe `fromJust` is applied only on `[Maybe a]` after `Nothing` values are filtered out. I will see if this can be done more succinctly.

Reply:
what is the issue here not able to understand i.e fixing mapMaybe to map instead? so we can't use maybe on a map?

Reply:
The problem here is we are apply a function on a list of Maybe's with `mapMaybe`. `mapMaybe` removes elements in the list which are `Nothing`:

```
Prelude> import Data.Maybe
Prelude Data.Maybe> mapMaybe (fmap (+2)) [Nothing, Just 3]
[5]
Prelude Data.Maybe> :t mapMaybe
mapMaybe :: (a -> Maybe b) -> [a] -> [b]
```

So in the above code, when we `zipWith` the `names` list we end up putting matching up the wrong header names with header values. @krisis' fix is to only use `map` and not filter the elements which are `Nothing` and handling the Maybe's after zipping.